src/App.vue

```diff
@@ -1,8 +1,132 @@
 <script lang="ts" setup>
+const toggleDrawer = () => {
+  const drawer = document.querySelector('.nav-links')
+  drawer?.classList.toggle('active')
+}
+
+const reloadPage = () => {
+  location.reload()
+}
+
+const scrollToElement = (id: string) => {
+  const drawer = document.querySelector('.nav-links')
+  drawer?.classList.toggle('active')
+
+  document.getElementById(id)?.scrollIntoView()
+}
 </script>
 
 <template>
+  <header h-12 md:h-20 lg:h-28 flex flex-row justify-between items-center gap-2 lg:gap-4 mx-0 class="header">
+    <button mx-4 md:mx-8 class="hamburger" :onclick="toggleDrawer">
+      <div class="bar" />
+    </button>
+
+    <div flex flex-row items-center gap-2>
+      <img w-8 md:w-12 lg:w-16 src="./assets/logo.png" alt="Logo">
+      <h1 text-white text-xl md:text-3xl font-bold>
+        超級智慧潛水
+      </h1>
+    </div>
+
+    <div w-16 />
+  </header>
```

Comment:
Suggest extracting this to a new component and use `slot` to insert into `App.vue` instead of directly write here.
https://vuejs.org/guide/components/slots.html#slots

Reply:
Addressed at f9c88a5 and following comments.